pkg/e2e/openshift/README.md

```diff
@@ -0,0 +1,57 @@
+## Overview
+
+This e2e tests provide a mechanism to test end-to-end the behavior of the components of the system (i.e Openshift) which relate in any manner to the machine API domain.
+
+Different testing suites with different primary purposes can be run here, e.g UX expectations, performance expectations, etc. 
+
+This e2e suites **must** run against any repo which **might** break the e2e holistic expected behaviour of the system related in any manner to the machine API domain:
+
+https://github.com/openshift/machine-api-operator
+
+https://github.com/openshift/cluster-autoscaler-operator
+
+https://github.com/openshift/cluster-api-provider-aws
+
+https://github.com/openshift/cluster-api-provider-libvirt
+
+https://github.com/openshift/autoscaler
+
+## Goals
+
+- Ensure consistent and reliable behavior of the system related in any manner to the machine API domain
+
+- Provide a reusable solution across any repo which might break the holistic expected behaviour of the system
+
+- Debuggability. Test suites should provide as detailed as possible reasons for the failure in its output
+
+## Non Goals
+
+- Validate behaviour of any component of the system completely unrelated to the machine API domain
+
+- Pre create Openshift/kubernetes clusters to run the expectations against it
+
+## Implementation
+
+Currently pure golang is used for running suite expectations but this can be moved to a framework eg. ginkgo as requirement rises
```

Comment:
`s/eg./e.g./`

Reply:
What's the benefit of the current implementation over ginkgo framework? We are already at a point where it's time to reuse ginkgo:
- to generate junit files
- to specify subset of tests to be run
- better readability of the code (everything else is part of framework initialization)
- community is already familiar with the technology
- support for various assertions we don't need to implement
- order of tests can by randomized before running
- tests can be run in parallel (which may discover potential flakes/bugs) compare to running them in series as we have now